### PR TITLE
unsafe methods to set name and values of ListSexp

### PR DESCRIPTION
**Pull request by daniellga**
Added 2 unsafe methods. This time I decided to make them public as it seems to be the better way in the case of `ListSexp`, since there are no constructors from slices (and I think it's better not to have them in this case).
I also signaled as unsafe other "unchecked" functions.
Please feel free to criticize and change the code at your will.

### CODE REVIEW

**Comment by yutannihilation** [comment]
Thank you! Looks good to me.

One advice is that you need three `/`s, not two to write the crate doc.

File: src/sexp/list.rs [hunk 1]
```diff
@@ -135,33 +141,51 @@ impl OwnedListSexp {
...
+        unsafe { self.set_name_unchecked(i, str_to_charsxp(k)?) };
+
         Ok(())
     }
 
+    /// # Safety
+    ///
+    /// The user has to assure bounds are checked.
+    #[inline]
+    pub unsafe fn set_name_unchecked(&mut self, i: usize, k: SEXP) {
```
Comment (eitsupi) [inline src/sexp/list.rs:174]: Is this implementation correct?
I could not figure out how to use it because `str_to_charsxp` is not public.

Comment (daniellga) [reply src/sexp/list.rs:174]: Hi! It has the same implementation as `set_name`. The difference is `set_name` checks bounds. So it should work fine.
```
    pub fn set_name(&mut self, i: usize, k: &str) -> crate::error::Result<()> {
        assert_len(self.len, i)?;

        unsafe { self.set_name_unchecked(i, str_to_charsxp(k)?) };

        Ok(())
    }

    /// # Safety
    ///
    /// Calling this method with an out-of-bounds index is undefined behavior.
    #[inline]
    pub unsafe fn set_name_unchecked(&mut self, i: usize, k: SEXP) {
        if let Some(names) = self.names.as_mut() {
            unsafe { names.set_elt_unchecked(i, k) };
        }
    }
```

`set_name_unchecked` accepts a `savvy::ffi::SEXP` as argument, so if you have a `&str` you would have to create a `SEXP` as he did with `str_to_charsxp()`. Copying his function and using to create a SEXP should work fine.

```
pub(crate) unsafe fn str_to_charsxp(v: &str) -> crate::error::Result<SEXP> {
    unsafe {
        // We might be able to put `R_NaString` directly without using
        // <&str>::na(), but probably this is an inevitable cost of
        // providing <&str>::na().
        if v.is_na() {
            Ok(savvy_ffi::R_NaString)
        } else {
            crate::unwind_protect(|| {
                Rf_mkCharLenCE(
                    v.as_ptr() as *const c_char,
                    v.len() as i32,
                    cetype_t_CE_UTF8,
                )
            })
        }
    }
}
```

Comment (eitsupi) [reply src/sexp/list.rs:174]: I get it. Thanks for the quick answer.

**Pull request merged**